cmd/kubeadm/app/phases/copycerts/copycerts.go

```diff
@@ -159,11 +160,26 @@ func createRBAC(client clientset.Interface) error {
 	})
 }
 
-func getSecretOwnerRef(client clientset.Interface, tokenID string) ([]metav1.OwnerReference, error) {
+func getSecretOwnerRef(client clientset.Interface, tokenID string, isDryRun bool) ([]metav1.OwnerReference, error) {
 	secretName := bootstraputil.BootstrapTokenSecretName(tokenID)
-	secret, err := client.CoreV1().Secrets(metav1.NamespaceSystem).Get(context.TODO(), secretName, metav1.GetOptions{})
-	if err != nil {
-		return nil, errors.Wrap(err, "error to get token reference")
+
+	var err error
+	var secret *v1.Secret
+
+	if isDryRun {
+		// If dry-running, just generate a fake bootstrap token Secret instead of doing a real query
+		secret = &v1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: metav1.NamespaceSystem,
+				Name:      secretName,
+				UID:       uuid.NewUUID(),
+			},
+		}
+	} else {
+		secret, err = client.CoreV1().Secrets(metav1.NamespaceSystem).Get(context.TODO(), secretName, metav1.GetOptions{})
+		if err != nil {
+			return nil, errors.Wrap(err, "error to get token reference")
```

Comment:
we can fix this typo

```suggestion
			return nil, errors.Wrap(err, "error getting token reference")
```